### PR TITLE
Add Git guide for team members

### PR DESCRIPTION
**Pull request by alihussain03**
Add Git guide first version

### CODE REVIEW

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md]: remove -> resolve

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md]: this conflicts with don't push straight to main

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md]: master -> main

File: GitGuide.md
Comment (rvsape) [inline GitGuide.md:35]: could possibly just delete branch after merge

**Comment by OpaJaa** [comment]
Seems simple enough guide. Don't know how familiar everyone is with git, so maybe there could be example cli commands for different tasks?

And could we have a common .gitignore to avoid some of the points in "Best practices"?

**Review by OpaJaa — COMMENTED** [review]
Reviewed. Comments in conversation.

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: Done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md:35]: Done

**Comment by alihussain03** [comment]
@OpaJaa i have added one link for common commands. it will be difficult to add all commands here and maybe it will be rework. Regarding gitignore file, as it will be js project so we can use this file and update accordingly https://gist.github.com/andreasonny83/b24e38b7772a3ea362d8e8d238d5a7bc

**Comment by OpaJaa** [comment]
> 

A link is ok. I was thinking like git checkout main, git branch cc-feature-1 / git checkout cc-feature-1 etc. Don't know if people even use cli or if git is well known for everyone. 

Also that .gitignore looks promising.

**Review by OpaJaa — DISMISSED** [review]

**Review by rvsape — DISMISSED** [review]

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: Isn't this flow now pushing straight to main branch? Should it be something like, clone repo, create new branch, implement stuff, commit and push to remote, create pull request, resolve merge conflicts. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github here is a good link how you can use github to resolve conflicts.

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md:14]: Clone repository / pull latest to main

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: I would suggest dropping "if everyone" to "if one person". What do others think?

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: Create pull request to main branch

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: members

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: others

File: GitGuide.md
Comment (bunjovis) [inline GitGuide.md]: Which convention? I suggest naming after features, for example dev-somecomponent, dev-bugfixes, dev-document.

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: no it is not pushing to main branch. i have updated the line 7.

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: i think it should be everyone. you can add limited persons as reviewers but every reviewer should approve that

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: done

File: GitGuide.md
Comment (alihussain03) [reply GitGuide.md]: this is what we can decide in meeting. btw it is article to read https://codingsight.com/git-branching-naming-convention-best-practices/

File: GitGuide.md
Comment (OpaJaa) [reply GitGuide.md]: I'd say not everyone, but maybe there could be some division on who reviews whose code. I mean like backend devs review code of other backend devs or something. And maybe the lead developer should review all code?  But after that everyone of those should accept the changes. There might be a lot of code to review if everyone should review and accept everything all the time.

Comment (bunjovis) [reply GitGuide.md]: Lets discuss this on thursdays meeting

File: GitGuide.md
Comment (bunjovis) [reply GitGuide.md]: lets discuss this on thursdays meeting

File: GitGuide.md
Comment (hannakortetjarvi) [inline GitGuide.md:9]: Some corrections to grammar:
1) It is* good to have
2) Some recommendations* are listed* down

File: GitGuide.md
Comment (hannakortetjarvi) [inline GitGuide.md:35]: branch*

**Review by olivena — DISMISSED** [review]

**Review by rvsape — DISMISSED** [review]

**Review by bunjovis — APPROVED** [review]

**Review by OpaJaa — APPROVED** [review]

**Pull request merged**
